frontend/src/cards/NewTaskCard.tsx

```diff
@@ -0,0 +1,181 @@
+import React, {ChangeEvent, useEffect, useState} from 'react';
+import './NewTaskCard.css';
+import {useNavigate} from 'react-router-dom';
+import axios from 'axios';
+import useTasks from '../hook/useTasks';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import {TaskModel} from "../model/TaskModel";
+import {format} from 'date-fns';
+
+function NewTaskCard() {
+    const [inputTaskName, setInputTaskName] = useState('');
+    const [inputCreator, setInputCreator] = useState('');
+    const [inputCategory, setInputCategory] = useState('');
+    const [createdate, setCreatedate] = useState<string>(getTodayDate());
+    const [selectedDate, setSelectedDate] = useState<Date | null>(null);
+    const [inputDescription, setInputDescription] = useState('');
+    const [inputAmoundOfPeople, setInputAmoundOfPeople] = useState<number | null>(null);
+    const navigate = useNavigate();
+    const {getAllTasks} = useTasks();
+    const handleDateChange = (date: Date | null) => {
+        setSelectedDate(date);
+    };
+
+    useEffect(() => {
+        setCreatedate(getTodayDate());
+    }, []);
+
+    function useTextInput(event: ChangeEvent<HTMLTextAreaElement>) {
+        const {name, value} = event.target;
+
+        switch (name) {
+            case 'taskName':
+                setInputTaskName(value);
+                break;
+            case 'creator':
+                setInputCreator(value);
+                break;
+            case 'category':
+                setInputCategory(value);
+                break;
+            case 'description':
+                setInputDescription(value);
+                break;
+            case 'amoundOfPeople':
+                setInputAmoundOfPeople(Number(value));
+                break;
+            default:
+                break;
+        }
+    }
+
+    // ...
+
+    function addNewTask() {
+        if (
+            inputTaskName.trim() === '' ||
+            inputCreator.trim() === '' ||
+            inputCategory.trim() === '' ||
+            selectedDate === null ||
+            inputDescription.trim() === '' ||
+            inputAmoundOfPeople === null
+        ) {
+            const alertMessage = document.getElementById('alert-message');
```

Comment:
you directly manipulate the DOM using document.getElementById.
Instead, consider using state to control the visibility of the alert message in a more React-like manner